crates/ecies/src/ffi/ecies.rs

```diff
@@ -0,0 +1,130 @@
+use core::ffi::c_size_t;
+use std::ffi::{c_char, c_int, c_uchar};
+
+use aes_gcm::aead::rand_core::SeedableRng;
+use cloudproof_cover_crypt::reexport::crypto_core::{
+    asymmetric_crypto::{
+        curve25519::{X25519KeyPair, X25519PrivateKey, X25519PublicKey},
+        ecies::{ecies_decrypt, ecies_encrypt, ecies_encrypt_get_overhead_size},
+        DhKeyPair,
+    },
+    CsRng, KeyTrait,
+};
+use cosmian_ffi_utils::{ffi_read_bytes, ffi_unwrap, ffi_write_bytes};
+
+#[no_mangle]
+pub unsafe extern "C" fn h_ecies_generate_key_pair(
+    public_key_ptr: *mut c_uchar,
+    public_key_len: *mut c_int,
+    private_key_ptr: *mut c_uchar,
+    private_key_len: *mut c_int,
+) -> c_int {
+    let mut rng = CsRng::from_entropy();
+    let key_pair: X25519KeyPair = X25519KeyPair::new(&mut rng);
```

Comment:
Is this up to date? I thought there was no `X25519KeyPair` in crypto_core.

Reply:
as discussed it will be done in next PR, thank you